python/src/uagents/setup.py

```diff
@@ -20,16 +21,26 @@ def fund_agent_if_low(wallet_address: str):
     Returns:
         None
     """
-    ledger = get_ledger()
+    ledger = get_ledger(agent._test)
```

Comment:
Let's add a `ledger` property to the agent so that we can just use `agent.ledger`.

Reply:
We shouldn't really use private class members outside the class. That's kind of the point of them and the reason that pylint check is useful. If we really need access to it, then we can make a property or make it public, but in this case, since there is no such thing as a mainnet faucet, then it only makes sense to use the testnet faucet here:
```
ledger = get_ledger(test=True)
```
But I think we should also log a warning if an agent who's ledger is set to mainnet tries to get funds, to make it clear that this function is only for adding funds on the testnet.

One more thing is that this is a pretty big change that will break a lot of people's code, so I think we should make it backwards compatible. If the argument passed is an address, then we should simply add testnet funds to that address.